test/base/GuardManager.spec.ts

```diff
@@ -0,0 +1,68 @@
+import hre from "hardhat";
+import { loadFixture } from "@nomicfoundation/hardhat-toolbox/network-helpers";
+import { expect } from "chai";
+import { SignerWithAddress } from "@nomicfoundation/hardhat-ethers/signers";
+import { getGuardWithPassingChecks } from "../utils/mockGuardBuilder";
+
+describe("GuardManager", async () => {
+    let deployer: SignerWithAddress, user1: SignerWithAddress;
+
+    before(async () => {
+        [deployer, user1] = await hre.ethers.getSigners();
+    });
+
+    async function deployContractsFixture() {
+        [deployer, user1] = await hre.ethers.getSigners();
+
+        const guardManager = await hre.ethers.deployContract("GuardManager", { signer: deployer });
+        const guard = await getGuardWithPassingChecks();
+
+        return { guardManager, guard };
+    }
+
+    it("Should emit GuardChanged event when a guard is enabled", async () => {
+        const { guardManager, guard } = await loadFixture(deployContractsFixture);
+        const guardAddress = await guard.getAddress();
+        expect(await guardManager.connect(user1).setGuard(guardAddress))
+            .to.emit(guardManager, "GuardChanged")
+            .withArgs(user1, guardAddress);
+    });
+
+    it("Should return correct guard address", async () => {
+        const { guardManager, guard } = await loadFixture(deployContractsFixture);
+        const guardAddress = await guard.getAddress();
+        await guardManager.connect(user1).setGuard(guardAddress);
+        expect(await guardManager.getEnabledGuard(user1.address)).to.be.equal(guardAddress);
+    });
+
+    it("Should return zero address if guard is not enabled", async () => {
+        const { guardManager } = await loadFixture(deployContractsFixture);
+        expect(await guardManager.getEnabledGuard(user1.address)).to.be.equal(hre.ethers.ZeroAddress);
+    });
+
+    it("Should return zero address if guard address is reset to zero address", async () => {
+        const { guardManager, guard } = await loadFixture(deployContractsFixture);
+
+        const guardAddress = await guard.getAddress();
+        expect(await guardManager.connect(user1).setGuard(guardAddress));
+        expect(await guardManager.connect(user1).setGuard(hre.ethers.ZeroAddress));
+        expect(await guardManager.getEnabledGuard(user1.address)).to.be.equal(hre.ethers.ZeroAddress);
+    });
+
+    it("Should revert if user attempts to set random address as guard", async () => {
```

Comment:
why would the contract even care about the address? does it mean a guard that's not in the registry?